PF_inversion_20190907.py

```diff
@@ -95,10 +98,33 @@
 
     assert False, "PF Inversion only implemented for 'data_type' 'ubc_grav' | 'ubc_mag' | 'ftmg' "
 
-if survey.std is None:
+# 0-level the data if required, d0 = 0 level
+d0 = 0.0
+if "subtract_mean" in list(input_dict.keys()) and input_dict["data_type"].lower() in ['ubc_mag', 'ubc_grav']:
+    subtract_mean = input_dict["subtract_mean"]
+    if subtract_mean:
+        d0 = np.mean(survey.dobs)
+        survey.dobs -= d0
+        print('Removed data mean: {0:.6g}'.format(d0))
+else:
+    subtract_mean = False
+
+# Update the specified data uncertainty
+if "new_uncert" in list(input_dict.keys()) and input_dict["data_type"].lower() in ['ubc_mag', 'ubc_grav']:
+    new_uncert = input_dict["new_uncert"]
+else:
+    new_uncert = [-1, -1]
```

Comment:
Can we do `new_uncert = False` instead? Cleaner

Reply:
How about ad18102? new_uncert isn't used anywhere else.